text/trace/0000-error_flagging.md

```diff
@@ -0,0 +1,86 @@
+# Error Flagging with Status Codes
+This proposal adds two status codes explicitly for use as overrides by the end user, and proposes a canonical mapping of semantic conventions to status codes. This clarifies how error reporting should work in OpenTelemetry.
+ 
+## Motivation
+Error reporting is a fundamental use case for distributed tracing. While we prefer that error flagging occurs within analysis tools, and not within instrumentation plugins, a number of currently supported analysis tools and protocols rely on the existence of an explicit error flag reported from instrumentation. In OpenTelmetry, the error flag is called "status codes."
+ 
+However, there is confusion over the mapping of semantic conventions to status codes, and concern over the subjective nature of errors. Which network failures count as an error? Are 404s an error? The answer is dependent on the situation.
+ 
+There is one major exception. Both application developers and operators have a deep understanding of what constitutes an error in their system. OpenTelemetry must provide a way for these users to control error flagging, and explicitly indicate both when a span should and should not be counted as an error.
+A second exception is supporting analysis tools which require explicit error flagging in the data which they receive. In this case, an operator must be able to apply an error flagging schema at some point during the OTLP data processing pipeline.
+ 
+## Explanation
+The following changes add several missing features required for proper error reporting, and are completely backwards compatible with OpenTelemetry today.
+ 
+### Status Codes
+The following status codes are added to our current  schema.
+ 
+* `DEFAULT` No status has been set. Any errors must be detected by the analysis tool. (This replaces `OK`.)
```

Comment:
Naming is hard... what is the default? the default is unspecified. Perhaps UNSPECIFIED could be used here?

Reply:
I would prefer NOT_SET. Then hopefully no one's gonna ask "can you set it explicitly to NOT_SET?". I would also prefer to explicitly state that this status code SHOULD NOT be set (or if possible, make sure at compile time that it cannot be set, e.g. by only having this enumeration value in the spec and protocol, but not in actual APIs).

Reply:
I agree with NOT_SET. But I would expect someone to ask for an `unsetStatusCode` sort of operation.  I'm trying to think of what use case there would be to unset the status (but not to use an override status).

Reply:
Heh, well given that it looks like we may move to a `user_override` flag instead of new statuses, the current default of `OK` starts to make sense again, as you can now tell the difference between the default and an explicit override from the user.